contracts/modules/PermissionManager/AutomatedBlackList.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity ^0.4.24;
+import "./GeneralPermissionManager.sol";
+import "../TransferManager/ITransferManager.sol";
+
+/**
+ * @title AutomateBlacklist
+ * @dev to automate blacklist process
+ */
+contract AutomateBlacklist is GeneralPermissionManager,ITransferManager {
+    
+    /**
+     * @dev structure for the blacklist period
+     */
+
+    struct BlacklistDetails {
+        uint256 startDate;
+        uint256 endDate;
+        uint256 repeats;
+    }
+ 
+     /** 
+     *@dev to check if msg.sender can add new entry
+     */
+    modifier isAllowed() {
+        require(msg.sender == securityToken || checkPermission(msg.sender,address(this),"1"));
+        _;
+    }
+  /// use to hold the different blacklist corresponds to name of blacklist type 
+    mapping(string => BlacklistDetails) blacklists;
+/// use to hold the blackListed address corresponds to the blacklist type
+    mapping(address => string) inverstorToBlacklist;
+/// use to get list of the addresses for a particular blacklist
+    mapping(string => address[]) blacklistToAddress;
+    
+    /**
+     * @dev to check if transfer can be made
+     */
+    function verifyTransfer(address _from, address _to, uint256 _amount, bool _isTransfer) public returns(Result) {
+
+        string storage typeOfBlacklist=inverstorToBlacklist[_from];
+        require(bytes(inverstorToBlacklist[_from]).length!=0);
+        uint256 repetitions = (now / (blacklists[typeOfBlacklist].endDate - blacklists[typeOfBlacklist].startDate + blacklists[typeOfBlacklist].repeats));
+        uint256 newStart = blacklists[typeOfBlacklist].startDate + ((blacklists[typeOfBlacklist].endDate - blacklists[typeOfBlacklist].startDate + blacklists[typeOfBlacklist].repeats) * repetitions);
+        uint256 newEnd = blacklists[typeOfBlacklist].endDate + ((blacklists[typeOfBlacklist].endDate - blacklists[typeOfBlacklist].startDate + blacklists[typeOfBlacklist].repeats) * repetitions);
+
+        if (now > newStart && now < newEnd){ return Result.INVALID;}
+        return Result.VALID;
+    }
+     /**
+      * @dev to add the new blacklist entry
+      * @param {string} _type
+      * @param {address} _investor
+      * @param {uint256} _start
+      * @param {uint256} _end
+      * @param {uint256} _repeat
+      */
+    function addEntry(string _type,address _investor,uint256 _start,uint256 _end,uint256 _repeat) isAllowed() public {
```

Comment:
I am not preferring this type of blacklist entry addition. 
Suppose there are 10 employees in a company and you are restricting all of them with the same date restrictions then it doesn't make sense to rewrite the state variables(`blacklists[_type]=BlacklistDetails(_start,_end,_repeat);`) again and again 

I prefer to add one function which only adds the blackList type extensively as I mentioned in the specs 
```function addBlackListType(uint256 _startDate, uint256 _endDate, bytes32 _name, uint256 _repeatPeriodInDays);```
and one function for the addition of the employees/affiliates corresponds to BlackList Type
```function addInvestorToBlackList(address _investor, bytes32 _blackListName);```

Have a look at the [lucidchart]( https://www.lucidchart.com/invitations/accept/c544a325-e74d-4e54-a360-71805a9f10ce) for exact requirements.